apps/table-frontend/src/app/components/from-research.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from 'react';
+import { getFile } from './get-file-info';
+import './from-research.css';
+import {
+  ParsedArticleParagraphStandalone,
+  ArticleParagraphBacksUpClaim,
+} from '@foodmedicine/interfaces';
+
+interface Paragraph {
+  url: string;
+  body: string;
+}
+
+export interface FromResearchProps {
+  fileName: string;
+}
+
+function LinkedShortParagraph(props: { paragraph: Paragraph }) {
+  function shortenText(text: string) {
+    return text.length > 400 ? text.substring(0, 400) + '...' : text;
+  }
+
+  return (
+    <p
+      className="linked-short-p"
+      onClick={() => window.open(props.paragraph.url, '_blank')}
+    >
+      {shortenText(props.paragraph.body)}
+    </p>
+  );
+}
+
+function getParagraphs(
+  parsedParagraphs: ParsedArticleParagraphStandalone[]
+): Paragraph[] {
+  return parsedParagraphs.map((parsedParagraph) => {
+    return {
+      url: parsedParagraph.head.xmlFullTextDownloadLink,
+      body: parsedParagraph.body,
+    };
+  });
+}
+
+export function FromResearch(props: FromResearchProps) {
+  const parsedParagraphs = getFile(props.fileName);
+  const paragraphs: Paragraph[] = getParagraphs(parsedParagraphs);
+  return (
+    <div>
+      {paragraphs.map((paragraph) => (
```

Comment:
have you tested that this .map function works as intended within returned html?

Reply:
Yep

Reply:
![Screenshot_2020-06-18_093812](https://user-images.githubusercontent.com/18239203/85027072-7d3a7980-b147-11ea-9131-b8eedac37d3f.jpg)

Reply:
Sorry I missed the screenshot in your original comments. Looks great! Really cool to see this all come together :)